File: java/kora/base-project/src/main/java/com/github/mroose/backend/config/client/BaseWebClientConfig.java
package com.github.mroose.backend.config.client;

import com.github.mroose.backend.dto.SampleDto;
import ru.tinkoff.kora.http.client.common.annotation.HttpClient;
import ru.tinkoff.kora.http.common.HttpMethod;
import ru.tinkoff.kora.http.common.annotation.HttpRoute;

import java.util.List;

@HttpClient
public interface BaseWebClientConfig {

    @HttpRoute(method = HttpMethod.GET, path = "/todos")
    List<SampleDto> getTodos();
}

File: kotlin/spring/base-project/src/main/kotlin/com/github/mroose/backend/config/client/BaseWebClientConfig.kt
package com.github.mroose.backend.config.client

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.client.WebClient

@Configuration
class BaseWebClientConfig {

    @Bean
    fun baseWebClient(): WebClient {
        return WebClient.create()
    }
}